libr/bin/format/pe/pe.c

```diff
@@ -3320,7 +3320,7 @@ struct r_bin_pe_section_t* PE_(r_bin_pe_get_sections)(struct PE_(r_bin_pe_obj_t)
 			}
 		} else {
 			memcpy (sections[j].name, shdr[i].Name, PE_IMAGE_SIZEOF_SHORT_NAME);
-			sections[j].name[PE_IMAGE_SIZEOF_SHORT_NAME - 1] = '\0';
+			sections[j].name[PE_IMAGE_SIZEOF_SHORT_NAME] = '\0';
```

Comment:
this causes an oobwrite error. ... err wtf who the hell did `        ut8 name[PE_IMAGE_SIZEOF_SHORT_NAME * 3];` in the .h??

Reply:
yeah that may not crash but it looks just wrong

Reply:
Agreed, but again, I did not investigate this personally. I'll ask the guy tomorrow.

Reply:
Okay, here is the answer from my colleague:

PE part:

As you said, it does not crash but leads to wrong section names.

This is an example that illustrate the problem:

before the patch:
```
12 0x00630400 156672 0x1406ae000 159744 m-r-x PAGEVRF
13 0x00656800  8704 0x1406d5000 12288 m-r-x PAGEHDL
14 0x00658a00 25600 0x1406d8000 28672 m-r-x PAGEBGF
15 0x00000000     0 0x1406df000 20480 m-rw- PAGEVRF_1
18 0x00674e00 48128 0x14070a000 49152 m-rw- PAGEVRF_2
```

after the patch:
```
12 0x00630400 156672 0x1406ae000 159744 m-r-x PAGEVRFY
13 0x00656800  8704 0x1406d5000 12288 m-r-x PAGEHDLS
14 0x00658a00 25600 0x1406d8000 28672 m-r-x PAGEBGFX
15 0x00000000     0 0x1406df000 20480 m-rw- PAGEVRFB
18 0x00674e00 48128 0x14070a000 49152 m-rw- PAGEVRFD
```

About the `ut8 name[PE_IMAGE_SIZEOF_SHORT_NAME * 3]`, the '*3' seems to be arbitrary and used to handle long section name: https://github.com/radare/radare2/blob/d8a32623e20f7766ab0f9172627c285214a9eb02/libr/bin/format/pe/pe.c#L3304
The code that parses long section name looks also wrong since it `memcpy` in the name `PE_IMAGE_SIZEOF_SHORT_NAME * 3` bytes of a buffer instead of read a null terminated string. That can also lead to section name shorter than it should.

source: www.osdever.net/documents/PECOFF.pdf sections 4 and 5.6.


PDB part:

I do not found any documentation that tell if PDB file support long section name and the code of the PDB does not support it anyway. So '*3' is not necessary since short section name are stored on 8 bytes max.